### PR TITLE
Eng job fixes

### PR DESCRIPTION
**Pull request by laurieboyes**

### CODE REVIEW

File: jobs/lib/label-remover.js [hunk 1]
```diff
@@ -46,7 +46,7 @@ class LabelRemover {
 				}
 				this.count += count;
 				console.log(`Removing label ${this.label} in progress. Removed from ${this.count.toLocaleString()} nodes`);
-				worker.metrics.count('label-removed', this.count);
+				worker.metrics.count('label-removed', count);
```
Comment (laurieboyes) [inline jobs/lib/label-remover.js]: :smile:

File: jobs/engagement.js [hunk 2]
```diff
@@ -93,17 +93,26 @@ function getUserEngagementQueries (userId) {
...
-		const userId = userStuff.userId;
-		const userEngagement = {};
-		userStuff.results.forEach(result => {
-			Object.assign(userEngagement, result[0]);
+	return Promise.all(batchResults.map(userStuff => {
+		return new Promise((resolve, reject) => {
+			const userId = userStuff.userId;
+			const userEngagement = {};
+			userStuff.results.forEach(result => {
+				Object.assign(userEngagement, result[0]);
```
Comment (keirog) [inline jobs/engagement.js]: One-liner police :rotating_light:

File: jobs/engagement.js [hunk 3]
```diff
@@ -93,17 +93,26 @@ function getUserEngagementQueries (userId) {
...
+			});
+
+			userEngagement.is_myft_user = isMyFtUser(userEngagement);
+
+			redisClient.hmset(userId + ':engagement', userEngagement, (err) => {
```
Comment (keirog) [inline jobs/engagement.js]: err

File: jobs/lib/all-users-queries.js [hunk 4]
```diff
@@ -117,24 +118,34 @@ class AllUsersQueries {
...
+						return this._run();
+					});
 
 			})
 			.catch(err => {
-				console.log(err);
 				worker.metrics.count('error');
+
+				if(++this.errorRetryCount <= 5) {
+					console.log(`Retrying for ${this.errorRetryCount}th time after error`, err);
```
Comment (keirog) [inline jobs/lib/all-users-queries.js]: th/st/nd but wevs

**Comment by keirog** [comment]
:+1:

File: jobs/lib/all-users-queries.js (hunk 4 above)
Comment (laurieboyes) [inline jobs/lib/all-users-queries.js]: :wink:

File: jobs/engagement.js [hunk 5]
```diff
@@ -93,17 +93,25 @@ function getUserEngagementQueries (userId) {
...
+			const userId = userStuff.userId;
+			const userEngagement = {};
+
+			userStuff.results.forEach(result => Object.assign(userEngagement, result[0]));
+
+			userEngagement.is_myft_user = isMyFtUser(userEngagement);
+
+			redisClient.hmset(userId + ':engagement', userEngagement, (err) => {
+				if(err) {
+					reject(err);
```
Comment (phamann) [inline jobs/engagement.js]: Do you want this reject?

**Pull request merged**